### PR TITLE
Feature/command line

### PR DESCRIPTION
**Pull request by mariofv**
**TLDR:  This branch was supposed to create a script that handles all the build process, but at the end it turned out to be a clean-up branch.**

Automatizing the build process implies importing all assets whithout using the engine at all. For this a new project/configuration is required, in order to run just the `ModuleResourceManagement` methods. Because this module is located inside the `Application` class, a lot of complexity arises:
- There are a lot of dependencies of the Application to be handled.
- If these dependencies are handled correctly, then the methods of `ModuleResourceManager`, and thus the new `Application`, should be encapsulated in a static library. Using the current soultion doesn't work at all, and creating a new Config that creates just the `.lib` doesn't either (because it will contain the Engine's `main` method, that will confuse the compiler on the BuildCreation project).

So, the BuildCreator cannot be created easily atm. The solution for this could be deattaching `ModuleResourceManagement` (and probably `ModuleFileSystem`, as atm is only used for resources) from the Engine inside a new built-in API,  similar to what Unity does. 

This way, both the Engine and the BuildCreator project can execute code from this API without touching other's code at all.

What's new:
- Restructuration of folders.
- Deletion of unnused Assets.
- All Binaries created in compilation by VisualStudio are stored in a new folder, `Binaries`

### CODE REVIEW

File: Engine/Main/Application.cpp [hunk 1]
```diff
@@ -62,6 +62,16 @@ Application::~Application()
...
+void Application::Import()
+{
+
+	resources->Init();
+	while (!resources->thread_comunication.finished_loading)
+	{
+		std::cout << "vblablabla porcentaje blablabla";
+	}
+}
+
```
Comment (yalania) [inline Engine/Main/Application.cpp]: Better remove this that is very cutre xD

Comment (mariofv) [reply Engine/Main/Application.cpp]: Ok! I totally forgot about it :D

**Review by Dartemiss — APPROVED** [review]
It works well for me!
